packages/cli/src/theme.ts

```diff
@@ -76,28 +106,65 @@ export async function buildTheme(options: Partial<BuildThemeOptions> = {}) {
   }
   if (!themeName) error('Please provide a theme name or path to a theme file.')
   green(`Locating ${themeName}...`)
-  const endpoint = options.api || DEFAULT_THEME_API
 
   const format = options.format || guessFormat()
-  const theme = themeName?.startsWith('./')
-    ? await generate(
-        await localTheme(themeName),
+
+  let theme = null
+  // if the theme is in the themesList then grab it from the api
+  const themeFromList = themes.find((t: any) => t.slug === themeName)
+
+  // check if there is a local file matching the provided theme name
+  const themeFilePath = determineFilePath(themeName) || themeName
+  const localThemeFile = existsSync(resolve(process.cwd(), themeFilePath))
+    ? await readFile(themeName, 'utf-8')
+    : null
+
+  if (themeFromList) {
+    info('Fetching theme from themes.formkit.com')
+    theme = await apiTheme(
+      themeName,
+      DEFAULT_THEME_API,
+      options.variables ?? '',
+      format === 'ts',
+      options.semantic ?? false
+    )
+  } else if (localThemeFile) {
+    info('Found local theme file')
+    theme = await generate(
+      await localTheme(themeName),
+      options.variables,
+      format === 'ts',
+      options.semantic,
+      themeName
+    )
+  } else {
+    // try to load theme from locally installed package
+    // using dynamic import
+    info('Fetching theme from local npm packages')
+    try {
+      const modulePath = resolve(process.cwd(), 'node_modules', themeName)
```

Comment:
I don’t think we want to actually look in the node modules here — instead we should rely on standard module resolution to perform the import — which is should do just from the "naked" module name.